projects/portal/src/app/models/wallets/wallet.application.service.ts

```diff
@@ -0,0 +1,200 @@
+import { ConnectWalletCompletedDialogComponent } from '../../views/dialogs/wallets/connect-wallet-completed-dialog/connect-wallet-completed-dialog.component';
+import { ConnectWalletStartDialogComponent } from '../../views/dialogs/wallets/connect-wallet-start-dialog/connect-wallet-start-dialog.component';
+import { UnunifiBackupMnemonicAndPrivateKeyWizardDialogComponent } from '../../views/dialogs/wallets/ununifi/ununifi-backup-mnemonic-and-private-key-wizard-dialog/ununifi-backup-mnemonic-and-private-key-wizard-dialog.component';
+import { UnunifiCreateWalletFormDialogComponent } from '../../views/dialogs/wallets/ununifi/ununifi-create-wallet-form-dialog/ununifi-create-wallet-form-dialog.component';
+import { UnunifiImportWalletWithMnemonicFormDialogComponent } from '../../views/dialogs/wallets/ununifi/ununifi-import-wallet-with-mnemonic-form-dialog/ununifi-import-wallet-with-mnemonic-form-dialog.component';
+import { UnunifiKeyFormDialogComponent } from '../../views/dialogs/wallets/ununifi/ununifi-key-form-dialog/ununifi-key-form-dialog.component';
+import { UnunifiSelectCreateImportDialogComponent } from '../../views/dialogs/wallets/ununifi/ununifi-select-create-import-dialog/ununifi-select-create-import-dialog.component';
+import { UnunifiSelectWalletDialogComponent } from '../../views/dialogs/wallets/ununifi/ununifi-select-wallet-dialog/ununifi-select-wallet-dialog.component';
+import { WalletType, StoredWallet } from './wallet.model';
+import { WalletService } from './wallet.service';
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class WalletApplicationService {
+  constructor(
+    private readonly walletService: WalletService,
+    private readonly dialog: MatDialog,
+    private snackBar: MatSnackBar,
+  ) {}
+
+  async connectWalletDialog(): Promise<void> {
+    const currentCosmosWallet = await this.walletService.getCurrentCosmosWallet();
+
+    const selectedWalletType = await this.openConnectWalletStartDialog();
+
+    if (!selectedWalletType) {
+      this.snackBar.open('Dialog was canceled!', 'Close');
+      return;
+    }
+
+    // Todo: After implementation, this should be removed
+    if (
+      selectedWalletType === WalletType.keplr ||
+      selectedWalletType === WalletType.keyStation ||
+      selectedWalletType === WalletType.ledger
+    ) {
+      this.snackBar.open('Selected Wallet is not supported yet!', 'Close');
+      return;
+    }
+
+    if (
+      selectedWalletType !== WalletType.ununifi &&
+      selectedWalletType !== WalletType.keplr &&
+      selectedWalletType !== WalletType.keyStation &&
+      selectedWalletType !== WalletType.ledger
+    ) {
+      this.snackBar.open('Invalid wallet type!', 'Close');
+      return;
+    }
+
+    if (selectedWalletType === WalletType.ununifi) {
```

Comment:
ブロック節に変えてifのネストを排除した方が良さげですけど、どうでしょうか？

Reply:
https://github.com/UnUniFi/web-apps/pull/32#discussion_r845710497
Code readability has improved with above link's fix. So, I think this is acceptable now, but how do you think about it?
(Personally, I think this will be fine as it matches the flow of the dialogs.)